Sileo/Backend/Package Manager/PackageListManager.swift

```diff
@@ -361,9 +361,9 @@ final class PackageListManager {
            !searchQuery.isEmpty {
             let search = searchQuery.lowercased()
             packageList.removeAll { package in
-                return [package.package, package.packageDescription, package.author, package.maintainer]
-                    .filter { $0?.lowercased().contains(search) ?? false }
-                    .isEmpty
+                // check if the user search term is in the package ID, description or in the author / maintainer name
+                let searchFields = [package.package, package.packageDescription, package.author, package.maintainer]
+                return !searchFields.contains { $0?.lowercased().localizedCaseInsensitiveContains(search) ?? false }
```

Comment:
Rather than inverting this condition I would personally change the line above to `packageList = packageList.filter { … }`.

Reply:
Would’ve done that but I was scared that may break something since this code is called in various places

Reply:
What’s wrong with removeAll? Wouldn’t that be the same in this context?

Reply:
Readability-wise I generally prefer code that avoids double negation

Reply:
Eh, doesn’t matter much